staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go

```diff
@@ -24,49 +24,63 @@ import (
 	"golang.org/x/time/rate"
 )
 
-type RateLimiter interface {
+type RateLimiter TypedRateLimiter[any]
+
+type TypedRateLimiter[T comparable] interface {
 	// When gets an item and gets to decide how long that item should wait
-	When(item interface{}) time.Duration
+	When(item T) time.Duration
 	// Forget indicates that an item is finished being retried.  Doesn't matter whether it's for failing
 	// or for success, we'll stop tracking it
-	Forget(item interface{})
+	Forget(item T)
 	// NumRequeues returns back how many failures the item has had
-	NumRequeues(item interface{}) int
+	NumRequeues(item T) int
 }
 
 // DefaultControllerRateLimiter is a no-arg constructor for a default rate limiter for a workqueue.  It has
 // both overall and per-item rate limiting.  The overall is a token bucket and the per-item is exponential
+//
+// Deprecated: Use DefaultTypedControllerRateLimiter instead.
```

Comment:
do we also deprecate `RateLimiter`? Why this symbol, but not many others in this change? (am just curious, no strong opinions. My gut feeling is that many of these are super-wide spread in the ecosystem).

Reply:
Just that I missed it. If there are no objections I would like to deprecate all the untyped variants. It will not cause any build failures, but if ppl use linters they will nudge them to start using the typed variants.